src/frontend/js/widgets/Dashboard/components/DashboardItem/Order/DashboardItemOrder.tsx

```diff
@@ -150,15 +83,22 @@ export const DashboardItemOrder = ({
 
   const intl = useIntl();
   const query = useCourseProduct(course.code, { productId: order.product });
+  const [contractFrameOpened, setContractFrameOpened] = useState(false);
+  const [contractLoading, setContractLoading] = useState(false);
   const product = query?.item?.product;
-  const needsSignature = order.contract && !order.contract.signed_on;
-
+  const needsSignature = orderNeedsSignature(order, product);
   const getRoutePath = getDashboardRoutePath(useIntl());
 
   return (
     <div className="dashboard-item-order">
-      {writable && !order.contract?.signed_on && (
-        <DashboardItemOrderContract order={order} product={product} />
+      {writable && needsSignature && (
```

Comment:
off this new feature review: 
I didn't like this naming "writable" when i had to work on these components.
We have two way to display this `DashboardItem`: page and list item or full and compact.
The fact that it's currently writable in page and not in list can be temporary, it can evolve.

Reply:
Yes we have a subject to discuss about here, plus it is starting to get a bit hard to read